docs/docs/clients/02-go-client.md

```diff
@@ -0,0 +1,300 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Account `alice` was initialized during `ignite chain serve`
+	accountName := "alice"
+
+	// Get account from the keyring
+	account, err := client.Account(accountName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	addr, err := account.Address(addressPrefix)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Define a message to create a post
+	msg := &types.MsgCreatePost{
+		Creator: addr,
+		Title:   "Hello!",
+		Body:    "This is the first post",
+	}
+
+	// Broadcast a transaction from account `alice` with the message
+	// to create a post store response in txResp
+	txResp, err := client.BroadcastTx(ctx, account, msg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Print response from broadcasting a transaction
+	fmt.Print("MsgCreatePost:\n\n")
+	fmt.Println(txResp)
+
+	// Instantiate a query client for your `blog` blockchain
+	queryClient := types.NewQueryClient(client.Context())
+
+	// Query the blockchain using the client's `PostAll` method
+	// to get all posts store all posts in queryResp
+	queryResp, err := queryClient.PostAll(ctx, &types.QueryAllPostRequest{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Print response from querying all the posts
+	fmt.Print("\n\nAll posts:\n\n")
+	fmt.Println(queryResp)
+}
+```
+
+The code above creates a standalone Go program that acts as a client to the
+`blog` blockchain. It begins by importing the required packages, including the
+general purpose Cosmos blockchain client and the `types` package of the `blog`
+blockchain.
+
+In the `main` function, the code creates a Cosmos client instance and sets the
+address prefix to "cosmos". It then retrieves an account named `"alice"` from
+the keyring and gets the address of the account using the address prefix.
+
+Next, the code defines a message to create a blog post with the title "Hello!"
+and body "This is the first post". It then broadcasts a transaction from the
+account "alice" with the message to create the post, and stores the response in
+the variable `txResp`.
+
+The code then instantiates a query client for the blog blockchain and uses it to
+query the blockchain to retrieve all the posts. It stores the response in the
+variable `queryResp` and prints it to the console.
+
+Finally, the code prints the response from broadcasting the transaction to the
+console. This allows the user to see the results of creating and querying a blog
+post on the `blog` blockchain using the client.
+
+To find out more about the `cosmosclient` package, you can refer to the Go
+package documentation for
+[`cosmosclient`](https://pkg.go.dev/github.com/ignite/cli/ignite/pkg/cosmosclient).
+This documentation provides information on how to use the `Client` type with
+`Options` and `KeyringBackend`.
+
+## Run the blockchain and the client
+
+Make sure your blog blockchain is still running with `ignite chain serve`.
+
+Install dependencies for your `blogclient`:
+
+```bash
+go mod tidy
```

Comment:
Running this before writing main.go is necessary for gopls to avoid compilation error. I don't know if this is the case under VSCode, which aslo uses gopls, but under vim that's for sure.